File: tests/CoreTests/Methods/Params.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.Methods
{
    [JsType(JsMode.Clr)]
    class Params
    {
        void alert(object obj)
        {
        }
        void foo()
        {
            var a = String.Format("{0} {1} {2} {3}", "a", "b", "c", "d");
            var b = String.Format("{0} {1} {2} {3}", "a", "b", "c");
            var c = String.Format("{0} {1} {2} {3}", "a", "b");
            var d = String.Format("{0} {1} {2} {3}", "a");        
            var e = String.Format("{0} {1} {2} {3}");
            if (CallMethodWithManyParams() != 0) 
                alert("params error"); //CRASHES
            if (CallMethodWithManyParams("aa") != 1) 
                alert("params error");
            if (CallMethodWithManyParams("a", "b") != 2) 
                alert("params error");
            if (CallMethodWithManyParams("a", "b", 1230) != 3) 
                alert("params error");
            if (CallMethodWithManyParams("a", "b", 1230, "sss") != 4) 
                alert("params error");

        }

        [JsMethod(NativeParams=false)]
        private static int CallMethodWithManyParams(params object[] arr)
        {
            return arr.Length;
        }
    }
}

File: tests/CoreTests/Methods/Params.js

if (typeof(JsTypes) == "undefined")
    var JsTypes = [];
var CoreTests$Methods$Params = {
    fullname: "CoreTests.Methods.Params",
    baseTypeName: "System.Object",
    staticDefinition: {
        CallMethodWithManyParams: function (arr){
            return arr.length;
        }
    },
    assemblyName: "CoreTests",
    Kind: "Class",
    definition: {
        ctor: function (){
            System.Object.ctor.call(this);
        },
        alert: function (obj){
        },
        foo: function (){
            var a = System.String.Format$$String$$Object$Array("{0} {1} {2} {3}", ["a", "b", "c", "d"]);
            var b = System.String.Format$$String$$Object$$Object$$Object("{0} {1} {2} {3}", "a", "b", "c");
            var c = System.String.Format$$String$$Object$$Object("{0} {1} {2} {3}", "a", "b");
            var d = System.String.Format$$String$$Object("{0} {1} {2} {3}", "a");
            var e = System.String.Format$$String$$Object$Array("{0} {1} {2} {3}", new Array(0));
            if (CoreTests.Methods.Params.CallMethodWithManyParams(new Array(0)) != 0)
                this.alert("params error");
            if (CoreTests.Methods.Params.CallMethodWithManyParams(["aa"]) != 1)
                this.alert("params error");
            if (CoreTests.Methods.Params.CallMethodWithManyParams(["a", "b"]) != 2)
                this.alert("params error");
            if (CoreTests.Methods.Params.CallMethodWithManyParams(["a", "b", 1230]) != 3)
                this.alert("params error");
            if (CoreTests.Methods.Params.CallMethodWithManyParams(["a", "b", 1230, "sss"]) != 4)
                this.alert("params error");
        }
    }
};
JsTypes.push(CoreTests$Methods$Params);

